homeassistant/components/light/__init__.py

```diff
@@ -292,17 +293,68 @@ class SetIntentHandler(intent.IntentHandler):
                 pars[ATTR_PROFILE] = Profiles.get_default(light.entity_id)
                 preprocess_turn_on_alternatives(pars)
                 turn_light_off, off_pars = preprocess_turn_off(pars)
+
             if turn_light_off:
-                await light.async_turn_off(**off_pars)
+                light_calls.append(light.async_turn_off(**off_pars))
             else:
-                await light.async_turn_on(**pars)
+                light_calls.append(light.async_turn_on(**pars))
+
+            if not light.should_poll:
+                continue
+
+            update_tasks.append(
+                light.async_update_ha_state(True))
+
+        if light_calls:
+            await asyncio.gather(*light_calls, loop=hass.loop)
```

Comment:
We had this when we first switched to asyncio, but most integrations could not handle this, and start erroring out. So we can't just change our code back again to it.  

We had a similar issue with updating, that we fixed by allowing platforms to specify the number of parallel commands (with 0 meaning unlimited).

Light is actually the odd one out, most entity components are running their services via the service helper, and hit [this code](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/service.py#L320), called from [here](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/service.py#L307-L311)